src/serlio/materials/MaterialCommand.cpp

```diff
@@ -163,6 +165,13 @@ MStatus MaterialCommand::redoIt() {
 
 MStatus MaterialCommand::undoIt() {
 	MStatus status;
+	MELScriptBuilder scriptBuilder;
+
+	//apply initial shading group to avoid other connected shading groups to disrupt undo process
+	scriptBuilder.setsUseInitialShadingGroup(fGeometryName.asWChar());
```

Comment:
in case of failed undo, do we need to also undo setting the initial shading group?

Reply:
Ideally, yes. 
However, I am not sure if there is a simple solution for that. We would need to store all assigned faceRanges and their corresponding shading group and reassign them in case of a failed undo...
I could write a follow-up ticket, but I am not sure how important this is, especially when the undo should ideally not fail in the first place..

Reply:
Do you want me to make any changes/ create a follow-up ticket or leave it as-is, given the arguments above?

Reply:
ok, sounds like a lot of work for a corner case... let's leave it.